controller.go

```diff
@@ -284,13 +284,17 @@ func syncPV(pv *PV) {
 			// NOTE: releasePV may either release the PV back into the pool or
 			// recycle it or do nothing (retain)
 
-			// HOWTO RELEASE A PV
-			pv.Status.Phase = Released
-			if err := CommitPVStatus(pv); err != nil {
-				// Status was not saved; we will fall back into the same
-				// condition in the next call to this method
-				return
+			// Make sure we don't overwrite previous failure of
+			// recycler/deleter.
+			if pv.Status.Phase != Failed {
```

Comment:
Hmm.. this makes me uneasy. We are making a decision based on Phase. Is there really no way to inspect a pv and realize that it is failed ?
I think first we should introduce setting phase to Failed into the pseudo code just to spell out what that means then worry about not overriding it.